numpy/core/src/multiarray/flagsobject.c

```diff
@@ -28,21 +29,20 @@ PyArray_NewFlagsObject(PyObject *obj)
     PyObject *flagobj;
     int flags;
 
-    if (obj == NULL) {
```

Comment:
I don't think we can make this change. First, `PyArray_NewFlagsObject` is part of the C-API, and second, it was already called twice in numpy with a `NULL` argument, indicating that perhaps that was not an uncommon usage.

That said, the old code could implement the logic in a cleaner way: `if ... else if ... else`.